src/dune_rules/melange_rules.ml

```diff
@@ -9,6 +9,33 @@ let make_js_name ~js_ext ~dst_dir m =
   let name = Melange.js_basename m ^ js_ext in
   Path.Build.relative dst_dir name
 
+let virtual_dst_dir_and_modules sctx ~target_dir ~implements ~scope =
```

Comment:
I based this function on `Virtual_rules.impl`, but it doesn't compute anything related to `foreign_objects`, and it retrieves the `dst_dir` information.

Not sure if there are opportunities to consolidate, it seemed cleaner to have a separate implementation, considering how specific Melange is.